src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AWS/SQS/ContextPropagation.cs

```diff
@@ -0,0 +1,47 @@
+// <copyright file="ContextPropagation.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SQS
+{
+    internal static class ContextPropagation
+    {
+        private const string SqsKey = "_datadog";
+
+        private static void Inject<TMessageRequest>(SpanContext context, IDictionary messageAttributes)
+        {
+            // Consolidate headers into one JSON object with <header_name>:<value>
+            StringBuilder sb = new();
+            sb.Append("{");
+            SpanContextPropagator.Instance.Inject(context, sb, ((carrier, key, value) => carrier.Append($"\"{key}\":\"{value}\",")));
+            sb.Remove(startIndex: sb.Length - 1, length: 1); // Remove trailing comma
+            sb.Append("}");
```

Comment:
It's very minor, but seeing as this change was performance reasons, I believe using `Append(char)` over `Append(string)` has some performance improvements, as well as using `AppendFormat()` instead of interpolation to avoid allocating an extra string.

```suggestion
            StringBuilder sb = new();
            sb.Append('{');
            SpanContextPropagator.Instance.Inject(context, sb, ((carrier, key, value) => carrier.AppendFormat("\"{0}\":\"{1}\",", key, value)));
            sb.Remove(startIndex: sb.Length - 1, length: 1); // Remove trailing comma
            sb.Append('}');
```

Reply:
Great suggestion, a quick benchmark shows even fewer allocations now! Fixed in 4ae4651a277ffb107d72e19425c5e800d1fe92b0

Reply:
For reference:

```
|                Method |     Mean |    Error |   StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------------------- |---------:|---------:|---------:|-------:|------:|------:|----------:|
|        ExistingMethod | 594.4 ns | 11.79 ns | 24.08 ns | 0.3138 |     - |     - |    1484 B |
| WithAndrewSuggestions | 553.9 ns | 11.01 ns | 26.60 ns | 0.1783 |     - |     - |     842 B |
```

Reply:
Uhh, actually fixed in 1ff312053519354e4683e32be2a12fdf5578b84d